lib/services/metadata_helper.dart

```diff
@@ -0,0 +1,156 @@
+      deviceModel = windowsInfo.deviceId;
+      deviceType = "PC"; // You can add more logic to determine the device type
+    } else {
+      throw UnsupportedError("Unsupported platform");
+    }
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      "osVersion": osVersion,
+      "deviceModel": deviceModel,
+      "deviceType": deviceType,
+    };
+  }
+}
+
+class AppInfo {
+  late String version;
+  late String originallyInstalledVersion;
+  late DateTime installationDate;
+  late List<Map<String, String>> updateHistory;
+
+  AppInfo();
+
+  Future<void> init() async {
+    PackageInfo packageInfo = await PackageInfo.fromPlatform();
+    version = packageInfo.version;
+    originallyInstalledVersion =
+        "1.0.0"; // This should be fetched from persistent storage
+    installationDate =
+        DateTime.now(); // This should be fetched from persistent storage
+    updateHistory = []; // This should be fetched from persistent storage
+  }
+
+  void addUpdateEvent(String newVersion) {
+    updateHistory.add({
+      "timestamp": DateTime.now().toIso8601String(),
+      "version": newVersion,
+    });
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      "version": version,
+      "originallyInstalledVersion": originallyInstalledVersion,
+      "installationDate": installationDate.toIso8601String(),
+      "updateHistory": updateHistory,
+    };
+  }
+}
+
+enum AddressType { LocalIp, LocalDomain, FullDomain }
+
+class ServerInfo {
+  late String serverVersion;
+  late String protocol;
+  late AddressType addressType;
+
+  ServerInfo();
+
+  Future<void> init() async {
+    // Fetch server information dynamically
+    serverVersion = "1.2.3"; // This should be fetched from the server
+    protocol = "HTTPS"; // This should be fetched from the server
+    addressType = AddressType
+        .FullDomain; // This should be determined based on the server address
+  }
+
+  Map<String, dynamic> toJson() {
+    return {
+      "serverVersion": serverVersion,
+      "protocol": protocol,
+      "addressType": addressType.toString(),
+    };
+  }
+}
+
+enum MetaDataType { AppInfo, ServerInfo, DeviceInfo, All }
+
+class MetaData {
+  late AppInfo appInfo;
+  late ServerInfo serverInfo;
+  late DeviceInfo deviceInfo;
+
+  MetaData() {
+    appInfo = getIt<AppInfo>();
+    serverInfo = getIt<ServerInfo>();
+    deviceInfo = getIt<DeviceInfo>();
+  }
+
+  Future<void> init() async {
+    await appInfo.init();
+    await serverInfo.init();
+    await deviceInfo.init();
+  }
+
+  Map<String, dynamic> toJson({MetaDataType? type}) {
+    switch (type) {
+      case MetaDataType.AppInfo:
+        return appInfo.toJson();
+      case MetaDataType.ServerInfo:
+        return serverInfo.toJson();
+      case MetaDataType.DeviceInfo:
+        return deviceInfo.toJson();
+      default:
+        return {
+          "appInfo": appInfo.toJson(),
+          "serverInfo": serverInfo.toJson(),
+          "deviceInfo": deviceInfo.toJson(),
+        };
```

Comment:
This doesn't generate JSON. The resulting string has no quotes at all. Please use the `@JsonSerializable` annotation and `toJSON` method like it's done in `finamp_models.dart`. You'll have to run code generation after adding the annotation. You can do that by running `dart run build_runner build --delete-conflicting-outputs`.